### PR TITLE
bets api implementation

### PR DESCRIPTION
**Pull request by koines**

### CODE REVIEW

File: lecture_4/02_bets_api/.env [hunk 1]
```diff
@@ -0,0 +1 @@
+SQLITE_DATABASE="../../lecture_3/03_project/db/bets.db"
```
Comment (vpintaric-axilis) [inline lecture_4/02_bets_api/.env]: Missing end-of-file newline.

File: lecture_4/02_bets_api/internal/api/controllers/controller.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+func (e *Controller) FetchSpecificId() gin.HandlerFunc {
+	return func(context *gin.Context) {
+		id := context.Param("id")
+
+		bet, err := e.betDtoService.GetByID(context, id)
+		if err != nil {
+			//context.String(http.StatusInternalServerError, "request could not be processed.")
+			context.String(http.StatusInternalServerError, err.Error())
```
Comment (vpintaric-axilis) [inline lecture_4/02_bets_api/internal/api/controllers/controller.go]: Leftover commented out code. Error handling across methods is not consistent - is this intentional?

File: lecture_4/02_bets_api/internal/api/controllers/controller.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+import (
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+// Controller implements handlers for web server requests.
+type Controller struct {
+	idValidator     BetDtoIdValidator
+	statusValidator BetDtoStatusValidator
```
Comment (vpintaric-axilis) [inline lecture_4/02_bets_api/internal/api/controllers/controller.go:11]: Does controller even use these two?

File: lecture_4/02_bets_api/internal/api/controllers/validators/betDtoIdValidator.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+type BetDtoIdValidator struct{}
+
+// NewBetDtoIdValidator creates a new instance of BetDtoIdValidator.
+func NewBetDtoIdValidator() *BetDtoIdValidator {
+	return &BetDtoIdValidator{}
+}
+
+// IdIsValid checks if event update is valid.
+// Id is not empty
+// Outcome is `lost`or `won`
```
Comment (vpintaric-axilis) [inline lecture_4/02_bets_api/internal/api/controllers/validators/betDtoIdValidator.go]: Comment doesn't really match the implementation (e.g. why is "outcome" mentioned here?). Also "Id is not empty" sounds like an unfinished sentence.

File: lecture_4/02_bets_api/internal/api/controllers/validators/betDtoStatusValidator.go [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+type BetDtoStatusValidator struct{}
+
+// NewBetDtoIdValidator creates a new instance of BetDtoIdValidator.
+func NewBetDtoStatusValidator() *BetDtoStatusValidator {
+	return &BetDtoStatusValidator{}
+}
+
+// IdIsValid checks if event update is valid.
+// Id is not empty
+// Outcome is `lost`or `won`
```
Comment (vpintaric-axilis) [inline lecture_4/02_bets_api/internal/api/controllers/validators/betDtoStatusValidator.go]: Same as with the other comment

File: lecture_4/02_bets_api/internal/infrastructure/sqlite/betRepository.go [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+	row.Next()
+
+	var customerId string
+	var status string
+	var selectionId string
+	var selectionCoefficient int
+	var payment int
+	var payoutSql sql.NullInt64
+
+	err = row.Scan(&id, &customerId, &status, &selectionId, &selectionCoefficient, &payment, &payoutSql)
```
Comment (vpintaric-axilis) [inline lecture_4/02_bets_api/internal/infrastructure/sqlite/betRepository.go]: How about:
```
	bet := storagemodels.Bet{}
	err := row.Scan(&bet.Id, &bet.CustomerId, ...)
	
	[...]

	return bet, nil
```

File: lecture_4/02_bets_api/internal/infrastructure/sqlite/betRepository.go [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+		var id string
+		var selectionId string
+		var selectionCoefficient int
+		var payment int
+		var payoutSql sql.NullInt64
+
+		err = row.Scan(&id, &customerId, &status, &selectionId, &selectionCoefficient, &payment, &payoutSql)
+		if err != nil {
+			return []storagemodels.Bet{}, err
+		}
```
Comment (vpintaric-axilis) [inline lecture_4/02_bets_api/internal/infrastructure/sqlite/betRepository.go]: Same refactoring suggestion

**Review by vpintaric-axilis — COMMENTED** [review]
Sorry for the late review!

File: lecture_4/02_bets_api/internal/api/controllers/controller.go (hunk 3 above)
Comment (koines) [reply lecture_4/02_bets_api/internal/api/controllers/controller.go:11]: Forgot to add those validators in code. Will do, thanks!

**Pull request merged**